pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -52,8 +55,70 @@ def __init__(self, runtime: Runtime, version: str, assembly: str, publish: bool,
         self.logger = runtime.logger
         self.working_dir = self.runtime.working_dir
         self.fail_count_name = f'count:build-sync-failure:{assembly}:{version}'
+        self.job_run = get_build_url()
+
+        self.slack_client = self.runtime.new_slack_client()
+        self.slack_client.bind_channel(f'openshift-{self.version}')
+
+    async def comment_on_assembly_pr(self, text_body):
```

Comment:
I think we can move the comment action in gen_assembly since pr object already have the context there and reduce the checking work. we only need much plugin in build_sync only need the result, and handle github operation in the same place.

Reply:
How would gen-assembly know about build-sync outcome? `comment_on_assembly_pr()` is called when build-sync completed, indicating either its success or failure. I don't see how this could be moved to gen-assembly

Reply:
it can wait and get the job run result https://github.com/openshift-eng/art-tools/blob/main/pyartcd/pyartcd/jenkins.py#L220 , if gen_assembly start with trigger build-sync parameter then it need to wait build-sync complete and do error alert or success comment there

Reply:
yeah that can be done, but it's a different approach. I still prefer to keep the two jobs loosely coupled, so I'd rather not have gen-assembly block and wait. Besides, unless I missed some important facts, it doesn't see relevant to have a blocking approach. The implemented code is already handling this by having build-sync infer PR info and comment there

Reply:
Besides, I think having build-sync comment about its own results is more appropriate than having gen-assembly do so. I don't want to establish a dependency relationship between the two